packages/cozy-search/src/components/Views/AssistantDialog.jsx

```diff
@@ -13,10 +13,21 @@ const AssistantDialog = () => {
   const { assistantState } = useAssistant()
   const { isMobile } = useBreakpoints()
   const navigate = useNavigate()
+  const location = useLocation()
+
   const { conversationId } = useParams()
 
   const onClose = () => {
-    navigate('..')
+    try {
+      const returnPath = new URLSearchParams(location.search).get('returnPath')
```

Comment:
you can use `useSearchParams` from react-router I think https://reactrouter.com/6.30.1/hooks/use-search-params `const [searchParams] = useSearchParams(); const returnPath = searchParams.get('returnPath')`